Dockerfile-ubuntu2004-dpdk1911

```diff
@@ -10,19 +12,24 @@ RUN ln -snf /usr/share/zoneinfo/$TZ /etc/localtime && echo $TZ > /etc/timezone
 RUN apt-get update && apt-get -y install \
   apt-utils \
   build-essential \
+  cmake \
   curl \
   git \
+  libnuma-dev \
+  libpcap-dev \
+  linux-virtual \
   tar \
   wget \
-  xz-utils \
-  libpcap-dev \
-  libnuma-dev \
-  cmake \ 
-  linux-headers-$(uname -r)
+  xz-utils
+
+# Create link to /lib/modules for this kernel
+RUN cd /lib/modules && \
+  ln -s *-generic `uname -r`
```

Comment:
Can you please explain this change? 🤔

Reply:
Docker just containairize the packages not the kernel, If you build the docker image on your host where the kernel is 5.15 or 5.17 then uname -r will return this version. Which Ubuntu 20.04 doesn't support.

So insted of doing this you can install a "Virtual Kernel Headers/Module" and hope it will compatible with all host machine running your Docker.

In this case this fix the build on recent kernel, and will not break running on Ubuntu 20.04 hosting machine.

Reply:
I was thinking about this... if we add this to the dockerfile the image will only use these kernel headers when running on the same host OS version it was built on. For example: if the image was built on Ubuntu 20.04 it will only use these headers if the image is running in Ubuntu 20.04 (hence the kernel headers have the same version as the symlink). But if the image is running in another OS or OS version it will not find this specific kernel headers version, so it will use the one on the host machine. Am I missing something here?

Reply:
No you are right, the module builtin in the Docker will be "compatible" with only Ubuntu 20.04 kernel.

So if you try to load a module on a different kernel example running on a Ubuntu 22.04 this docker, it may not work.

But normally you don't load kernel module from a Docker.
Docker should only use userspace tools, so either the host has to load the kernel module by itself or the better is to have a VM.

Anyway it's better to force this module to be compatible with this Ubuntu 20.04 virtual kernel and not the host machine kernel as it may result in different docker regarding where the Dockerfile is built from.

Reply:
It seems linux-headers are not required when building with meson on 20.11, or maybe meson automatically disable the kernel module if the headers are not found

Reply:
ok sure let's leave it as is. We anyway use these images in the CI which always runs a 20.04 VM